src/crypto/algorithms/megolm.ts

```diff
@@ -1298,7 +1298,9 @@ class MegolmDecryption extends DecryptionAlgorithm {
 
         if (res === null) {
             // We've got a message for a session we don't have.
-            //
+            // try and get the missing key from the backup first
+            this.crypto.backupManager.queryKeyBackupRateLimited(event.getRoomId(), content.session_id).catch(() => {});
```

Comment:
In order to reduce unneeded to_device traffic, on EA if we have a positive reply from the backup we don't call requestKeysForEvent. Is it possible to do the same here?

Reply:
Not trivially. If we naively `await` this operation, we get stuck in a deadlock because the key backup query operation includes importing the keys, which includes re-attempting decryption of the messages, which means waiting for the current decryption attempt to finish. So we can't use, or even know, the results of the backup query during the current decryption attempt, without a bunch of messy re-architecting. In order to still avoid sending key requests, we'd need to add more complicated state tracking to check whether we've already tried the key backups on a previous decryption attempt :/